bpf/bpf_netdev.c

```diff
@@ -507,7 +507,7 @@ static __always_inline int do_netdev_encrypt_fib(struct __ctx_buff *ctx,
 {
 	int ret = 0;
 
-#ifdef HAVE_FIB_LOOKUP
+#ifdef BPF__PROG_TYPE_sched_cls__HELPER_bpf_fib_lookup
```

Comment:
Ok, so the tests are currently failing because bpftool always defines these helper macros. They have value 1 or 0 depending on whether the helper is available or not. So this line and other lines checking for helper availability need to be changed to use `#if` instead of `#ifdef`. The same is not true for the `HAVE_` macros which bpftool does not define if the feature isn't available.

Reply:
I also discovered that bpftool provides the following macro:

```
/*** eBPF helper functions ***/
/*
 * Use HAVE_PROG_TYPE_HELPER(prog_type_name, helper_name)
 * to determine if <helper_name> is available for <prog_type_name>,
 * e.g.
 *      #if HAVE_PROG_TYPE_HELPER(xdp, bpf_redirect)
 *              // do stuff with this helper
 *      #elif
 *              // use a workaround
 *      #endif
 */
#define HAVE_PROG_TYPE_HELPER(prog_type, helper)        \
        BPF__PROG_TYPE_ ## prog_type ## __HELPER_ ## helper
```

I used it now.